File: DemoMethods/Indexes/CompaniesAndCountry.cs
using System.Linq;
using DemoMethods.Entities;
using Raven.Client.Indexes;

namespace DemoMethods.Indexes
{
    public class CompaniesAndCountry : AbstractIndexCreationTask<Company>
    {
        public class Result
        {
            public Company Company { get; set; }
            public string Address_Country { get; set; }
        }

        public CompaniesAndCountry()
        {
            Map = companies =>
                from company in companies
                select new
                {
                    Company = company,
                    Address_Country = company.Address.Country
                };
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/indexes/CompaniesAndCountry.java
package net.ravendb.demo.indexes;

import com.mysema.query.annotations.QueryEntity;
import net.ravendb.client.indexes.AbstractIndexCreationTask;
import net.ravendb.demo.entities.Company;

public class CompaniesAndCountry extends AbstractIndexCreationTask {

    @QueryEntity
    public static class Result {
        private Company company;
        private String address_Country;

        public String getAddress_Country() {
            return address_Country;
        }

        public void setAddress_Country(String address_Country) {
            this.address_Country = address_Country;
        }

        public Company getCompany() {
            return company;
        }

        public void setCompany(Company company) {
            this.company = company;
        }
    }

    public CompaniesAndCountry() {
        map = "from company in docs.Companies " +
                "select new { " +
                "  Company = company, " +
                "  Address_Country = company.Address.Country " +
                "}";
    }
}
